application/packages/python/ftrs_common/utils/jwt_auth.py

```
@@ -0,0 +1,159 @@
+import json
+import os
+import uuid
+from functools import cache
+from time import time
+from typing import Dict, Optional
+
+import boto3
+import jwt
+import requests
+from botocore.exceptions import ClientError
+
+
+class JWTAuthenticator:
+    def __init__(
+        self,
+        environment: Optional[str] = None,
+        region: str = "eu-west-2",
+        secret_name: Optional[str] = None,
+    ) -> None:
+        self.environment = environment or os.environ.get("ENVIRONMENT", "local")
+        self.region = region or os.environ.get("AWS_REGION", "eu-west-2")
+        self.custom_secret_name = secret_name
+
+    def get_jwt_credentials(self) -> Dict[str, str]:
+        if self.environment == "local":
+            return self._get_local_credentials()
+        return self._get_aws_credentials()
+
+    def _get_local_credentials(self) -> Dict[str, str]:
+        required_vars = [
+            "LOCAL_API_KEY",
+            "LOCAL_PRIVATE_KEY",
+            "LOCAL_KID",
+            "LOCAL_TOKEN_URL",
+        ]
+        missing_vars = [var for var in required_vars if not os.environ.get(var)]
+
+        if missing_vars:
+            raise JWTCredentialsError(missing_vars, "environment variables")
+
+        return {
+            "api_key": os.environ["LOCAL_API_KEY"],
+            "private_key": os.environ["LOCAL_PRIVATE_KEY"],
+            "kid": os.environ["LOCAL_KID"],
+            "token_url": os.environ["LOCAL_TOKEN_URL"],
+        }
+
+    def _get_aws_credentials(self) -> Dict[str, str]:
+        if self.custom_secret_name:
```

Recognise I'm proposing a negative check which isn't optimum but if we don't have custom_secret_name as it can be None we still progress to get the client etc and expect the failure in line 55. 
Should be fail faster i.e. 

```
 if not self.custom_secret_name:
        raise JWTCredentialsError(["secret_name"], "configuration")
    
  secret_name = self.custom_secret_name
```